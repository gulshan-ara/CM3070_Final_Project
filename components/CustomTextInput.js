// import libraries
import { StyleSheet, Text, TextInput, View } from "react-native";
import React from "react";

const CustomTextInput = ({ label, value, onChangeText }) => {
	// render view
	return (
		<View style={styles.container}>
			{/* Custom label for text input field */}
			<Text style={styles.label}>{label} :</Text>
			{/* Rendering the text input */}
			<TextInput
				style={styles.inputContainer}
				value={value} // sets the value of input field
				placeholder="Type here" // placeholder text to show initially
				onChangeText={onChangeText} // handles input change
			/>
		</View>
	);
};

export default CustomTextInput;

const styles = StyleSheet.create({
	container: {
		marginVertical: 5,
		marginHorizontal: 10,
		flexDirection: "row",
	},
	label: {
		fontWeight: "bold",
		fontSize: 18,
		margin: 5,
		paddingHorizontal: 5,
		flex: 1.5,
	},
	inputContainer: {
		flex: 2.5,
		marginHorizontal: 5,
		paddingHorizontal: 8,
		alignItems: "center",
		borderBottomWidth: 1,
	},
});
