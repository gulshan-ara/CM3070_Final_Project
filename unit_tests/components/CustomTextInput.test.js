/** This files contains unit tests for the custom text input component */

// import libraries
import React from "react";
import { render, fireEvent } from "@testing-library/react-native";

// import component file
import CustomTextInput from "../../components/CustomTextInput";

/** unit test - 1
 *  The purpose of this unit test is to check whether the component is rendered properly or not.
 */
describe("CustomTextInput", () => {
	it('render correctly text input component', () => {  
		const TextInputComponent = render(<CustomTextInput />).toJSON();
		expect(TextInputComponent).toMatchSnapshot();
	});

	it("renders CustomTextInput correctly", () => {
		// mocking the onChangeText function
		const onChangeText = jest.fn();
		// dummy values to pass as props
		const textInputValue = "Text Input Value";
		const textInputLabel = "Text Input Label";

		// rendering the component
		const { getByText, getByPlaceholderText } = render(
			<CustomTextInput
				onChangeText={onChangeText}
				value={textInputValue}
				label={textInputLabel}
			/>
		);

		// Assertions of successful render of the text input label
		const labelTest = getByText(`${textInputLabel} :`);
		expect(labelTest).toBeTruthy();

		// Assertions of successful render of the text input placeholder
		const placeHolderTextTest = getByPlaceholderText(`Type here`);
		expect(placeHolderTextTest).toBeTruthy();
	});

	/** unit test - 2
	 *  The purpose of this unit test is to check whether the component functions properly onChange text or not.
	 */
	it("fires onChangeText when text is typed", () => {
		// mocking the onChangeText function
		const onChangeText = jest.fn();

		// rendering the custom text input component
		const { getByPlaceholderText } = render(
			<CustomTextInput onChangeText={onChangeText} value="" label="" />
		);

		// storing the placeholder text in a variable
		const placeHolderText = getByPlaceholderText("Type here");

		// making an event where text is changed on the input field
		fireEvent.changeText(
			placeHolderText,
			"Changed Text after completing the event"
		);

		// Assertion of calling the mock function the typed text on text input field
		expect(onChangeText).toHaveBeenCalledWith(
			"Changed Text after completing the event"
		);
	});
});
