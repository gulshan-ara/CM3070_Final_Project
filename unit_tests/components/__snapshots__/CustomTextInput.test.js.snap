// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTextInput render correctly text input component 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "marginHorizontal": 10,
      "marginVertical": 5,
    }
  }
>
  <Text
    style={
      {
        "flex": 1.5,
        "fontSize": 18,
        "fontWeight": "bold",
        "margin": 5,
        "paddingHorizontal": 5,
      }
    }
  >
     :
  </Text>
  <TextInput
    placeholder="Type here"
    style={
      {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "flex": 2.5,
        "marginHorizontal": 5,
        "paddingHorizontal": 8,
      }
    }
  />
</View>
`;
